tests/test_terminal_bond_maps.py

```diff
@@ -0,0 +1,170 @@
+    states = [Interval(np.random.rand(), 1 + np.random.rand()) for _ in range(50)]
+
+    # generate test points up to within eps of interval bounds,
+    # (eps slightly > 0 to avoid spurious <=, >= assertion errors near bounds...)
+    eps = 1e-8
+
+    def construct_map(src: Interval, dst: Interval):
+        return partial(interval_map, src_lb=src.lower, src_ub=src.upper, dst_lb=dst.lower, dst_ub=dst.upper)
+
+    # for each pair of states, compute f, f_inv on a bunch of points, assert self-consistency
+    for src in states:
+        for dst in states:
+            f = construct_map(src, dst)
+            f_inv = construct_map(dst, src)
+
+            # x in src
+            xs = np.linspace(src.lower + eps, src.upper - eps, 1000)
+            np.testing.assert_array_less(src.lower, xs, err_msg="x not in support of src!")
+            np.testing.assert_array_less(xs, src.upper, err_msg="x not in support of src!")
+
+            # y=f(x) in dst
+            ys = vmap(f)(xs)
+            assert ys.shape == xs.shape
+            np.testing.assert_array_less(dst.lower, ys, err_msg="y not in support of dst!")
+            np.testing.assert_array_less(ys, dst.upper, err_msg="y not in support of dst!")
+
+            # x_=f_inv(f(x))
+            xs_ = vmap(f_inv)(ys)
+            np.testing.assert_array_less(src.lower, xs_, err_msg="f_inv(f(x)) not in support of src!")
+            np.testing.assert_array_less(xs_, src.upper, err_msg="f_inv(f(x)) not in support of src!")
+            np.testing.assert_allclose(xs_, xs, err_msg="f_inv(f(x)) != x!")
+
+
+# utility functions for vacuum test system
+
+
+def collect_samples(mol):
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+    AllChem.EmbedMolecule(mol)
+    samples = generate_ligand_samples(1000, mol, ff, 300.0, 2022)[0][:, 0]
+    return samples - samples[:, 0, np.newaxis]  # center first atom, for ease of visualization
+
+
+def get_hb_params(mol):
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+    params, bond_idxs = ff.hb_handle.parameterize(mol)
+    return params, bond_idxs
+
+
+def get_vacuum_u_fxn(mol, temperature=300.0):
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+    U_fxn = jit(VacuumState(mol, ff).U_full)
+    kBT = BOLTZ * temperature
+
+    def u_fxn(xs):
+        return vmap(U_fxn)(xs) / kBT
+
+    return u_fxn
+
+
+def test_on_methane():
+    """Test TerminalBondMaps on methane <-> halogen-substituted methane"""
+    np.random.seed(2022)
+
+    parser_params = Chem.SmilesParserParams()
+    parser_params.removeHs = False
+
+    src_mol = Chem.MolFromSmiles("C([H])([H])([H])([H])", parser_params)
+    dst_mol = Chem.MolFromSmiles("C(Br)(Cl)(F)(I)", parser_params)
+    AllChem.EmbedMolecule(src_mol)
+    AllChem.EmbedMolecule(dst_mol)
+
+    # define src and dst states
+    u_src = get_vacuum_u_fxn(src_mol)
+    src_params, src_bond_idxs = get_hb_params(src_mol)
+    src_state = TerminalMappableState.from_harmonic_bond_params(src_bond_idxs, src_params)
```

Comment:
Do any angle terms change in this example?

Reply:
Yes -- this is why standard deviation of mapped works in this example is not 0.

Reply:
Right, can you add this as a comment below

Reply:
Added comment in https://github.com/proteneer/timemachine/pull/801/commits/9130b4d23fb0bf8b0d4814c239314556f6e7d882